k8sutils/pkg/instrumentation_instance/status.go

```diff
@@ -13,77 +13,44 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-type InstrumentationInstanceConfig struct {
-	healthy                  *bool
-	identifyingAttributes    []odigosv1.Attribute
-	nonIdentifyingAttributes []odigosv1.Attribute
-	message                  string
-	reason                   string
-}
-
 type InstrumentationInstanceOption interface {
-	applyInstrumentationInstance(InstrumentationInstanceConfig) InstrumentationInstanceConfig
-}
-
-type fnOpt func(InstrumentationInstanceConfig) InstrumentationInstanceConfig
-
-func (o fnOpt) applyInstrumentationInstance(c InstrumentationInstanceConfig) InstrumentationInstanceConfig {
-	return o(c)
+	applyInstrumentationInstance(odigosv1.InstrumentationInstanceStatus) odigosv1.InstrumentationInstanceStatus
 }
 
-func WithHealthy(healthy *bool) InstrumentationInstanceOption {
-	return fnOpt(func(c InstrumentationInstanceConfig) InstrumentationInstanceConfig {
-		c.healthy = healthy
-		return c
-	})
-}
+type fnOpt func(odigosv1.InstrumentationInstanceStatus) odigosv1.InstrumentationInstanceStatus
```

Comment:
What is this fnOpt means?
I need you to elaborate me about `updateInstrumentationInstanceStatus` proc in general

Reply:
renamed to `updateInstrumentationInstanceStatusOpt`